internal/report/report.go

```diff
@@ -309,7 +309,10 @@ func printTopProto(w io.Writer, rpt *Report) error {
 	}
 	functionMap := make(functionMap)
 	for i, n := range g.Nodes {
-		f := functionMap.FindOrAdd(n.Info)
+		f, added := functionMap.FindOrAdd(n.Info)
+		if added {
+			out.Function = append(out.Function, f)
+		}
 		flat, cum := n.FlatValue(), n.CumValue()
 		l := &profile.Location{
```

Comment:
Doesn't have to be in this PR, but I wonder if locations should be de-duped like functions?

Reply:
Would we expect the that locations to be de-duplicated when the graph is constructed?

Reply:
I've checked with a number of large profiles and have not found one where locations a repeated (though I've not constructed any profile specifically for this purpose).

I've also confirmed that there is one node in the graph per location in the profile ( code confirming this [here](https://github.com/google/pprof/blob/master/internal/graph/graph.go#L520)).

So duplication of locations shouldn't be possible, (and if we see duplicated locations in the -topproto output, then the locations would be duplicated in pprof's internal proto representation, so the problem should be fixed elsewhere.)